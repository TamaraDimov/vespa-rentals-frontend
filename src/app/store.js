import { configureStore } from '@reduxjs/toolkit';
import userSlice from '../redux/reducers/userSlice';
import motorcycleReducer from '../redux/reducers/motorcycleSlice';
import reservationReducer from '../redux/reducers/reservationSlice';

const store = configureStore({
  reducer: {
    motorcycle: motorcycleReducer,
    reservation: reservationReducer,
    user: userSlice,
  },
});

export default store;
