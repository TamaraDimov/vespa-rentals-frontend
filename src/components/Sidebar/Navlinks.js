import React from 'react';
import { NavLink } from 'react-router-dom';
import links from './links';

// eslint-disable-next-line
const NavLinks = ({ handleLinkClick }) => (
  <div className="nav-links">
    {links.map((link) => {
      const { text, id, path } = link;
      return (
        <NavLink
          to={path}
          className={({ isActive }) => (isActive ? 'nav-link active' : 'nav-link')}
          key={id}
          onClick={() => {
            handleLinkClick();
          }}
          end
        >
          {text}
        </NavLink>
      );
    })}
  </div>
);

export default NavLinks;
