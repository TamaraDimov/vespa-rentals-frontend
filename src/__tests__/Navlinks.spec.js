import { create } from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../app/store';
import Navlinks from '../components/Sidebar/Navlinks';

describe('Navlinks component', () => {
  test('renders correctly', () => {
    const component = create(
      <Provider store={store}>
        <Router>
          <Navlinks />
        </Router>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
