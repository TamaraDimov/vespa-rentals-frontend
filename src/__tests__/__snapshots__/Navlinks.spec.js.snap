// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navlinks component renders correctly 1`] = `
<div
  className="nav-links"
>
  <a
    className="nav-link"
    href="/motorcycles"
    onClick={[Function]}
  >
    Models
  </a>
  <a
    className="nav-link"
    href="/reservations/new"
    onClick={[Function]}
  >
    Book a ride
  </a>
  <a
    className="nav-link"
    href="/reservations"
    onClick={[Function]}
  >
    My Reservations
  </a>
  <a
    className="nav-link"
    href="/motorcycles/new"
    onClick={[Function]}
  >
    ADD Motorcycle
  </a>
  <a
    className="nav-link"
    href="/motorcycles/delete-motorcycles"
    onClick={[Function]}
  >
    DELETE Motorcycle
  </a>
</div>
`;
